### PR TITLE
[Snyk] Fix for 5 vulnerabilities

### PR DESCRIPTION
**Pull request by sestinj**
snyk-top-banner

#### Snyk changed the following file(s):

- `manual-testing-sandbox/requirements.txt`

<details>
<summary>⚠️ <b>Warning</b></summary>

```
typer 0.3.2 has requirement click<7.2.0,>=7.1.1, but you have click 8.1.8.
thinc 8.0.17 requires murmurhash, which is not installed.
thinc 8.0.17 requires preshed, which is not installed.
thinc 8.0.17 requires cymem, which is not installed.
spacy 3.1.2 requires murmurhash, which is not installed.
spacy 3.1.2 requires preshed, which is not installed.
spacy 3.1.2 requires cymem, which is not installed.
qtconsole 5.4.4 requires pyzmq, which is not installed.
notebook 6.5.7 requires pyzmq, which is not installed.
jupyter-server 1.24.0 requires pyzmq, which is not installed.
jupyter-console 6.6.3 requires pyzmq, which is not installed.
jupyter-client 7.4.9 requires pyzmq, which is not installed.
ipykernel 6.16.2 requires pyzmq, which is not installed.
exceptiongroup 1.3.1 has requirement typing-extensions>=4.6.0; python_version < "3.13", but you have typing-extensions 3.10.0.2.
```

</details>

#### Breaking Change Risk

Merge Risk: High

> **Notice:** This assessment is enhanced by AI.

---

> [!IMPORTANT]
>
> - Check the changes in this PR to ensure they won't cause issues with your project.
> - Max score is 1000. Note that the real score may have changed since the PR was raised.
> - This PR was automatically created by Snyk using the credentials of a real user.
> - Some vulnerabilities couldn't be fully fixed and so Snyk will still find them when the project is tested again. This may be because the vulnerability existed within more than one direct dependency, but not all of the affected dependencies could be upgraded.

---

**Note:** _You are seeing this because you or someone else with access to this repository has authorized Snyk to open fix PRs._

For more information: 
🧐 [View latest project report](https://app.snyk.io/org/continue-dev-inc.-default/project/19fa7992-fc8c-4050-aa18-2600afc162f8?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr)
📜 [Customise PR templates](https://docs.snyk.io/scan-using-snyk/pull-requests/snyk-fix-pull-or-merge-requests/customize-pr-templates?utm_source=github&utm_content=fix-pr-template)
🛠 [Adjust project settings](https://app.snyk.io/org/continue-dev-inc.-default/project/19fa7992-fc8c-4050-aa18-2600afc162f8?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr/settings)
📚 [Read about Snyk's upgrade logic](https://docs.snyk.io/scan-with-snyk/snyk-open-source/manage-vulnerabilities/upgrade-package-versions-to-fix-vulnerabilities?utm_source=github&utm_content=fix-pr-template)

---

**Learn how to fix vulnerabilities with free interactive lessons:**

🦉 [Directory Traversal](https://learn.snyk.io/lesson/directory-traversal/?source&#x3D;registry-oss-fix-pr&amp;loc&#x3D;fix-pr)
🦉 [Server-side Request Forgery (SSRF)](https://learn.snyk.io/lesson/ssrf-server-side-request-forgery/?source&#x3D;registry-oss-fix-pr&amp;loc&#x3D;fix-pr)
🦉 [Regular Expression Denial of Service (ReDoS)](https://learn.snyk.io/lesson/redos/?source&#x3D;registry-oss-fix-pr&amp;loc&#x3D;fix-pr)

[//]: # 'snyk:metadata:{"breakingChangeRiskLevel":"high","FF_showPullRequestBreakingChanges":true,"FF_showPullRequestBreakingChangesWebSearch":false,"customTemplate":{"variablesUsed":[],"fieldsUsed":[]},"dependencies":[{"name":"keras","from":"2.6.0","to":"3.12.3"},{"name":"nltk","from":"3.6.3","to":"3.10.0"}],"env":"prod","issuesToFix":["SNYK-PYTHON-KERAS-18507704","SNYK-PYTHON-NLTK-18507646","SNYK-PYTHON-NLTK-18507677","SNYK-PYTHON-NLTK-18507687","SNYK-PYTHON-NLTK-18507692"],"prId":"4f8942d3-307f-4cd3-b4d1-c68e4b0257f3","prPublicId":"4f8942d3-307f-4cd3-b4d1-c68e4b0257f3","packageManager":"pip","priorityScoreList":[641,721,746,721,721],"projectPublicId":"19fa7992-fc8c-4050-aa18-2600afc162f8","projectUrl":"https://app.snyk.io/org/continue-dev-inc.-default/project/19fa7992-fc8c-4050-aa18-2600afc162f8?utm_source=github&utm_medium=referral&page=fix-pr","prType":"fix","templateFieldSources":{"branchName":"default","commitMessage":"default","description":"default","title":"default"},"templateVariants":["pr-warning-shown","priorityScore"],"type":"auto","upgrade":[],"vulns":["SNYK-PYTHON-KERAS-18507704","SNYK-PYTHON-NLTK-18507646","SNYK-PYTHON-NLTK-18507677","SNYK-PYTHON-NLTK-18507687","SNYK-PYTHON-NLTK-18507692"],"patch":[],"isBreakingChange":false,"remediationStrategy":"vuln"}'

### CODE REVIEW

**Comment by sestinj** [comment]
Merge Risk: High

This upgrade involves a major, high-risk update to `keras` and a medium-risk update to `nltk`.  
  
### keras@2.6.0 → keras@3.12.3  
**Risk: HIGH**  
  
The upgrade from Keras 2 to Keras 3 is a major rewrite with significant breaking changes. Keras 3 has been redesigned as a multi-backend framework that can run on top of TensorFlow, JAX, and PyTorch.  
  
**Key Breaking Changes:**  
*   **Package Imports:** Code must be migrated from `tensorflow.keras` to the new standalone `keras` package. For example, `from tensorflow.keras import layers` must be changed to `from keras import layers`.  
*   **Backend Configuration:** A backend must be explicitly configured (e.g., via the `KERAS_BACKEND` environment variable) before importing Keras.  
*   **API Changes:** To achieve multi-backend compatibility, code using TensorFlow-specific operations (e.g., `tf.nn.relu`) should be replaced with backend-agnostic alternatives like `keras.ops.relu`.  
*   **Model Saving:** The default model-saving format has changed to `.keras` v3. While older models can often be loaded, re-saving them in the new format is recommended.  
  
**Recommendation:**  
This is a significant migration that requires careful code modification and testing. Developers must review the official migration guide to update import paths and replace any backend-specific code. Due to the scale of these changes, this upgrade should be handled in a dedicated effort.  
  
### nltk@3.6.3 → nltk@3.10.0  
**Risk: MEDIUM**  
  
This upgrade spans several minor versions and includes environment and behavioral changes.  
  
**Key Changes:**  
*   **Python Version Support:** NLTK 3.7, included in this range, drops support for Python 3.6.  
*   **Behavioral Change:** As of version 3.6.5, METEOR evaluation now requires pre-tokenized input, which may affect existing implementations using this metric.  
*   **Enhancements:** The versions include numerous bug fixes, performance improvements, and support for newer Python versions like 3.10 and 3.11.  
  
**Recommendation:**  
Verify that your project's runtime environment is newer than Python 3.6. If you use METEOR for evaluation, ensure your input is pre-tokenized to avoid issues.

> **Notice 🤖:** This content was augmented using **artificial intelligence**. AI-generated content **may** contain **errors** and should be **reviewed** for accuracy before use.

[//]: # (snyk:breakability:{"publicId":"9539ad8d-4626-44bc-b7f6-5513a8d02997"})
